### PR TITLE
Migrate DescriptiveWithBasedOnSubjectSpec from spek to kotest 

### PR DESCRIPTION
**Pull request by jakubriegel**
Refers to https://github.com/robstoll/atrium/issues/1194

______________________________________
I confirm that I have read the [Contributor Agreements v1.0](https://github.com/robstoll/atrium/blob/main/.github/Contributor%20Agreements%20v1.0.txt), agree to be bound on them and confirm that my contribution is compliant.

### CODE REVIEW

File: misc/atrium-specs/build.gradle.kts [hunk 1]
```diff
@@ -24,6 +24,7 @@ kotlin {
                 apiWithExclude("io.kotest:kotest-runner-junit5:$kotestVersion")
                 // necessary in order that intellij sees the io.kotest symbols (runner-junit5 actually already depends on it)
                 apiWithExclude("io.kotest:kotest-framework-api:$kotestVersion")
+                implementation("org.jetbrains.kotlinx:kotlinx-coroutines-core-jvm:1.3.9")
```
Comment (jakubriegel) [inline misc/atrium-specs/build.gradle.kts:27]: I'm sure if this should be done this way in Atrium's dependencies setup. Feel free to suggest better approach.

Without adding coroutines explicitly as implementation here, gradle is not able to compile the code.

**Comment by jakubriegel** [comment]
@robstoll If review will take more time, is it possible, that we add `HACKTOBERFEST-ACCEPTED` label here?

File: misc/atrium-specs/build.gradle.kts (hunk 1 above)
Comment (robstoll) [reply misc/atrium-specs/build.gradle.kts:27]: All fine, I can still improve after merge

**Review by robstoll — APPROVED** [review]

**Comment by robstoll** [comment]
@jakubriegel looks good to me, thanks for the contribution

**Pull request merged**